### PR TITLE
checking for nil before asserting

### PR DESCRIPTION
**Pull request by dominicm**

### CODE REVIEW

**Comment by raphael** [comment]
This is great! thank you.

File: goagen/gen_app/generator_test.go [hunk 1]
```diff
@@ -493,7 +493,9 @@ func MountWidgetController(service goa.Service, ctrl WidgetController) {
 	mux := service.ServeMux()
 	h = func(c *goa.Context) error {
 		ctx, err := NewGetWidgetContext(c)
-		ctx.Payload = ctx.RawPayload().(Collection)
+		if rawPayload := ctx.RawPayload(); rawPayload != nil {
+			ctx.Payload = rawPayload.(Collection)
+		}
```
Comment (moorereason) [inline goagen/gen_app/generator_test.go]: Shouldn't this come after the `err != nil` if-block?  For example:

```go
ctx, err := NewGetWidgetContext(c)
if err != nil {
	return goa.NewBadRequestError(err)
}
if rawPayload := ctx.RawPayload(); rawPayload != nil {
	ctx.Payload = rawPayload.(Collection)
}
```

File: goagen/gen_app/writers.go [hunk 2]
```diff
@@ -528,7 +528,9 @@ func Mount{{.Resource}}Controller(service goa.Service, ctrl {{.Resource}}Control
 {{$res := .Resource}}{{$ver := .Version}}{{range .Actions}}{{$action := .}}	h = func(c *goa.Context) error {
 		ctx, err := New{{.Context}}(c)
 {{if not $ver.IsDefault}}		ctx.APIVersion = service.Version("{{$ver.Version}}").VersionName()
-{{end}}{{if .Payload}}		ctx.Payload = ctx.RawPayload().({{gotyperef .Payload nil 1}})
+{{end}}{{if .Payload}}		if rawPayload := ctx.RawPayload(); rawPayload != nil {
+				ctx.Payload = rawPayload.({{gotyperef .Payload nil 1}})
+		}
```
Comment (moorereason) [inline goagen/gen_app/writers.go]: Same issue here.

File: goagen/gen_app/generator_test.go (hunk 1 above)
Comment (raphael) [inline goagen/gen_app/generator_test.go]: Good point, thanks!

**Pull request merged**